File: src/main/java/red/tetracube/smartHub/SmartHubMain.java
package red.tetracube.smartHub;

import io.quarkus.runtime.Quarkus;
import io.quarkus.runtime.QuarkusApplication;
import io.quarkus.runtime.annotations.QuarkusMain;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@QuarkusMain
public class SmartHubMain {

    public static void main(String... args) {
        Quarkus.run(SmartHubApp.class, args);
    }

    public static class SmartHubApp implements QuarkusApplication {

        private static final Logger LOGGER = LoggerFactory.getLogger(SmartHubApp.class);

        @Override
        public int run(String... args) {
            LOGGER.info("Smart Hub is started, notifying components");
            Quarkus.waitForExit();
            return 0;
        }
    }
}

File: src/main/kotlin/red/tetracube/smarthub/SmartHubMain.kt
package red.tetracube.smarthub

import io.quarkus.runtime.Quarkus
import io.quarkus.runtime.QuarkusApplication
import io.quarkus.runtime.annotations.QuarkusMain
import org.slf4j.LoggerFactory
import red.tetracube.smarthub.iot.SmartBrokerHub

@QuarkusMain
class SmartHubMain {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            Quarkus.run(SmartHubApp::class.java, *args)
        }
    }

    class SmartHubApp(private val smartBrokerHub: SmartBrokerHub) : QuarkusApplication {

        private val logger = LoggerFactory.getLogger(SmartHubApp::class.java)

        @Throws(Exception::class)
        override fun run(vararg args: String): Int {
            logger.info("Smart Hub is started, notifying components")
            smartBrokerHub.setupMqttConnection()
            Quarkus.waitForExit()
            return 0
        }
    }
}